src/function/maybe_changed_after.rs

```diff
@@ -369,20 +411,22 @@ where
                 // Conditionally specified queries
                 // where the value is specified
                 // in rev 1 but not in rev 2.
-                VerifyResult::Changed
+                VerifyResult::changed()
+            }
+            QueryOrigin::FixpointInitial if old_memo.may_be_provisional() => {
+                VerifyResult::changed()
             }
-            QueryOrigin::FixpointInitial if old_memo.may_be_provisional() => VerifyResult::Changed,
             QueryOrigin::FixpointInitial => VerifyResult::unchanged(),
```

Comment:
@carljm I'm not 100% sure about the two fixpoint arms here.

Reply:
I'm not either :) I don't have a memory of why this is this way -- maybe it was needed in some much-earlier version of fixpoint iteration?

Locally it seems that replacing both arms with this doesn't fail any tests, and it feels more clearly correct (but I'd want to also check it for perf regression on ty):

```
QueryOrigin:FixpointInitial => VerifyResult::changed(),
```

Reply:
This makes me wonder if we even need this extra variant or if we could just use `Derived` everywhere because we also have `revisions.maybe_changed_after` to identify provisional values.